tests/builtin_functions.py

```diff
@@ -8,19 +8,22 @@ def test_any(self):
     self.assertOnlyIn(((2, 5), (3, 0)), detect("any()"))
 
   def test_basestring(self):
-    self.assertOnlyIn(((2, 3), (3, 0)), detect("basestring()"))
+    self.assertOnlyIn((2, 3), detect("basestring()"))
 
   def test_bin(self):
     self.assertOnlyIn(((2, 6), (3, 0)), detect("bin()"))
 
+  def test_callable(self):
+    self.assertOnlyIn(((2, 0), (3, 2)), detect("callable()"))
+
   def test_classmethod(self):
     self.assertOnlyIn(((2, 2), (3, 0)), detect("classmethod()"))
 
   def test_enumerate(self):
     self.assertOnlyIn(((2, 3), (3, 0)), detect("enumerate()"))
 
   def test_file(self):
-    self.assertOnlyIn(((2, 2), (3, 0)), detect("file()"))
+    self.assertOnlyIn((2, 0), detect("file()"))
```

Comment:
This is a tricky thing. The [`file` builtin function](https://docs.python.org/2/library/functions.html#file) is new in 2.2, but [file objects](https://docs.python.org/2/library/stdtypes.html#file-objects) has always existed in Python 2 since 2.0. The rule `"file.xreadlines": ((2, 1), None)` will contradict with `"file": ((2, 2), None)`.

Reply:
Also, in the current implementation of vermin, I guess rule matching is based on a "module name-member name chain match" rather than type deduction right? (I know that type deduction could be hard.) So `open(...).xreadlines` will not match the `"file.xreadlines": ((2, 1), None)` rule, but `file(...).xreadlines` will.

Reply:
That's a good point. Yes, it's using heuristics because type deduction isn't easy and I also cannot execute any code since it can have side-effects.

Let's keep it at 2.0 but then it would be great if the rule (in rules.py) as annotated with this reasoning in short.